dbt/include/athena/macros/materializations/seeds/helpers.sql

```diff
@@ -23,9 +32,9 @@
             {{ adapter.quote_seed_column(column_name, quote_seed_column) }} {{ type }} {%- if not loop.last -%}, {% endif -%}
         {%- endfor -%}
     )
-    stored as parquet
```

Comment:
So this means now the seed tables are not stored as parquet but as JSON format ?

Reply:
I think that's the case. Maybe we should still support for parquet.
In general the datasets pushed via seeds should be relatively small, but parquet is more optimal

Reply:
I would be in favour of supporting parquet.

Reply:
@nicor88 @Jrmyy I'm on your side parquet would be ideal, I tried and I can try again if you all want (maybe I was missing something the first time). 

From what I can tell, parquet would add a lot of complexity to this adapter for imho no real benefit, definitely given that seed files are fairly small. The performance gain we'd get from parquet vs complexity it would add to this adapter is what ultimately led me to use json. `agate.Table` doesn't support parquet out of the box which would mean adding `pyarrow` as a dependency and doing a whole bunch of type inference and conversion; We could also add `pandas` as a dependency (large dep) that could do some of that type inference & conversion for us. We'd also have to transfer `config.column_types` types when defined and convert them to their appropriate `pyarrow` data type regardless of if we're using `pyarrow` or `pandas` and that get's especially difficult if the seed files have complex data structures like a column that has json data. When using `json` the `column_types` just simply flow into the seed helpers macro and athena does the rest for us.

Reply:
You are right, the conversion json-parquet can lead to heavy lib complexity that is not the core of this adapter.
Also again good point on the size, seeds are generally small static files, so all good from my side, maybe we should be sure to make it explicit in the release notes that this could lead to some breaking change for some people...(not really I think that data is pruned and table re-created as json serialization)

Reply:
@Jrmyy what do you think about the above?

Reply:
@henriblancke do you mind to ask other opinion about the topic in db-athena in slack?
I've being using this feature for really really small files.

Reply:
If we want to go for a support of larger seed files, I would suggest revamping the whole seed strategy with some config. Because for small seed files, we don't want to upload file to S3 since it cas be an additional cost while not necessary.

Now, considering for the larger files, I would suggest not transforming to JSON and use the default CSV format because right now, with your implementation, you are loading the data in cache and it could be tricky for very large files. This is a suggestion I have:

- Load the csv in S3, without opening it, if possible
- Create the table matching the location of the uploaded file. By the way you can create tables with csv files like this:
```sql
CREATE EXTERNAL TABLE IF NOT EXISTS table_tmp (
               columns
               )
               ROW FORMAT DELIMITED
               FIELDS TERMINATED BY ","
               LINES TERMINATED BY "\n"
               LOCATION 's3://location/'
               TBLPROPERTIES (
               'skip.header.line.count' = '1'
               );
```
- Optional, to be discussed, create the final table as select * from this temp table
- Drop the csv and the tmp table

The optional step can be handy if we want to provide custom config like support for iceberg files if very very big seed files, partitioning, etc...

Reply:
I tried to keep this as simple as possible. The reason I didn't go with `csv` is because of type inference. `OpenCSVSerde` is not good at casting timestamps and inferring correct data types. But maybe I missed something?

Not sure I'm following with the `in cache` logic, I'm writing the `agate.Table` to a temporary file and then upload it to s3. Is there a way to circumvent dbt loading the seed file in memory and into an `agate.Table`? 

The idea here was not to support huge files (that you most likely can't even check into git without LFS and probably is a dbt anti pattern). It was to support slightly larger files than what the athena query char limit imposes. We have some csv seed files that are only a few 100 rows but are wide and can't be loaded because of that char limit.

Reply:
I'm a bit struggling with what is the best here, the implementation proposed by @henriblancke is actually pretty lean, only cons is that it's using JSON, a the same time would be nice to have parquet even for small files, as it's columnar and is way more efficient.

Here a proposal based on what @Jrmyy suggested combined with this proposed solution:
* upload the csv to s3
* read the header of the csv 
   ```
   import csv
   with open('data.csv') as csv_file:

       csv_reader = csv.reader(csv_file, delimiter = ',')
       header = csv.reader(csv_file, delimiter = ',')
    ```
* create an tmp external table that point to the csv - all columns can be treated as string
* create the final table as parquet with explicit casting (if provided by the user), sees gives the possibility to specify data types.
* drop the tmp external csv table
* remove the s3 csv file from s3

Reply:
You gain speed and cost saving, mostly on selective colums queries. Couple of mbs became couple of kbs if you select specific colums.
@Jrmyy what do you think on the above?

Reply:
I think this will work. It might be worth a try ! 🚀 
The only concern I have is about this:
> read the header of the csv

But if dbt only support csv with headers this is perfect.

Reply:
> read the header of the csv
I don't think that is something natively from dbt. the snippet above uses the csv library, and as csv reader use generators I expect that doesn't read the all file in memory.

Reply:
@nicor88 @Jrmyy just pushed a change that implements parquet, let me know what you all think of the approach. Thanks!